quickshop-bukkit/src/main/java/com/ghostchu/quickshop/menu/history/MainPage.java

```diff
@@ -124,7 +120,16 @@ public void handle(final PageOpenCallback callback) {
         //header icon
         final Shop shop = shops.get(0);
         final String world = (shop.getLocation().getWorld() != null)? shop.getLocation().getWorld().getName() : "World";
-        final String shopName = shop.getShopName() == null? world + " " + shop.getLocation().getBlockX() + ", " + shop.getLocation().getBlockY() + ", " + shop.getLocation().getBlockZ() : shop.getShopName();
+        final Component shopName;
+
+        if (shop.getShopName() != null) {
+          shopName = QuickShop.getInstance().text().of("history.shop.header-icon-shop-name", shop.getShopName()).forLocale();
+        } else {
+          shopName = QuickShop.getInstance().text().of("history.shop.header-icon-shop-empty-name", world, shop.getLocation().getBlockX(), shop.getLocation().getBlockY(), shop.getLocation().getBlockZ()).forLocale();
+        }
+
+        Component shopType = QuickShop.getInstance().text().of("shop-type." + shop.getShopType().name().toLowerCase(Locale.ROOT)).forLocale();
```

Comment:
This can be labeled as final.

Reply:
https://github.com/QuickShop-Community/QuickShop-Hikari/blob/hikari/.contributing/contributing.md